prov/rxd/src/rxd_ep.c

```diff
@@ -743,6 +784,110 @@ static int rxd_dg_cq_open(struct rxd_ep *rxd_ep, enum fi_wait_obj wait_obj)
 	return ret;
 }
 
+int rxd_epaddr_hash_insert(struct rxd_ep *ep, void *addr, 
+                 fi_addr_t *rxd_addr)
+{
+    struct rxd_epname_entry *entry = NULL;
+    
+    HASH_FIND(hh, ep->ep_rxdaddr_hash, addr, RXD_NAME_LENGTH, entry);
+    if (entry != NULL) {                                                          
+        return -FI_EALREADY;
+    }else {
+        entry = malloc(sizeof(struct rxd_epname_entry));
+        if (!entry)
+            return -FI_ENOMEM;        
+        assert(rxd_addr);
+        memset(entry->addr, 0, RXD_NAME_LENGTH);
+        memcpy(entry->addr, addr, RXD_NAME_LENGTH);
+        entry->rxd_addr = *rxd_addr;
+        HASH_ADD(hh, ep->ep_rxdaddr_hash, addr, RXD_NAME_LENGTH, entry);
+    }
+    return 0;
+    
+}
+
+struct rxd_peer* rxd_get_peer_entry(struct rxd_ep *ep, void *addr, size_t addrlen) 
+{
+    struct rxd_epname_entry *entry = NULL;
+    struct rxd_peer *peer_entry;
+    fi_addr_t rxd_addr; 
+    uint8_t tmp_addr[RXD_NAME_LENGTH];
+
+    memset(tmp_addr, 0, RXD_NAME_LENGTH);
+    memcpy(tmp_addr, addr, addrlen);
+    HASH_FIND(hh, ep->ep_rxdaddr_hash, tmp_addr, RXD_NAME_LENGTH, entry);
+    if (entry != NULL) {
+        peer_entry = ofi_bufpool_get_ibuf(ep->peer_pool.pool,
+                        entry->rxd_addr);
```

Comment:
in this case let's just have `return ofi_bufpool_get_ibuf(...)` and then outdent the code below in the else